python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -230,6 +228,8 @@ def makeTableSpec(cls, datasetIdColumnType: type) -> ddl.TableSpec:
                 ddl.FieldSpec(name="file_size", dtype=BigInteger, nullable=True),
             ],
             unique=frozenset(),
+            indexes=[tuple(["dataset_id", "path"]), tuple(["path"]),
+                     tuple(["dataset_id", "component", "path"])],
```

Comment:
Does this last `(dataset_id, component, path)` index yield a measured improvement, too?  I _think_ that could remove the need to actually look in the table when querying for the `path` given the `dataset_id` and `component`, as all the information would be in the index, but the primary key should already be sufficient to still do that in `O(log N)` time but with more overhead, and this is getting to be a lot of indexes.

Reply:
I added it because it seemed like the complement to the first one. From a quick test `path` alone works fine.